#DPATCHLEVEL=0
# Change FILL constants to enum
# Clean up fill_mask handling
--- src/libxosd/xosd.c~	2004-08-02 14:01:08.000000000 +0200
+++ src/libxosd/xosd.c	2004-08-02 17:31:03.000000000 +0200
@@ -79,12 +79,6 @@
 #define SLIDER_SCALE 0.8
 #define SLIDER_SCALE_ON 0.7
 
-//draw mask constants
-#define FILL_NONE    0
-#define FILL_OUTLINE 1
-#define FILL_SHADOW  2
-#define FILL_FACE    4
-
 const char *osd_default_font =
   "-misc-fixed-medium-r-semicondensed--*-*-*-*-c-*-*-*";
 static const char *osd_default_colour = "green";
@@ -92,14 +86,9 @@ static const char *osd_default_colour = 
 // const char* osd_default_font="adobe-helvetica-bold-r-*-*-10-*";
 // const char* osd_default_font="-adobe-helvetica-bold-r-*-*-10-*";
 
-
-
-typedef enum
-{ LINE_blank, LINE_text, LINE_percentage, LINE_slider } line_type;
-
 typedef struct
 {
-  line_type type;
+  enum { LINE_blank, LINE_text, LINE_percentage, LINE_slider } type;
 
   char *text;
   int length;
@@ -164,7 +153,7 @@ struct xosd
   int timeout;                  /* delta time */
   struct timespec timeout_time; /* Next absolute timeout */
 
-  int fill_mask;
+  enum { FILL_OUTLINE = 1, FILL_SHADOW = 2, FILL_FACE = 4 } fill_mask;
 };
 
 /** Global error string. */
@@ -296,17 +285,8 @@ expose_line(xosd * osd, int line)
   assert(osd);
   FUNCTION_START(Dfunction);
 
-  osd->fill_mask = FILL_NONE;
-  if (!osd->shadow_offset && !osd->outline_offset) {
-    osd->fill_mask = FILL_FACE;
-  } else {
-    if (osd->shadow_offset && !osd->outline_offset) {
-      osd->fill_mask = FILL_SHADOW;
-    } else {
-      if (osd->outline_offset)
-        osd->fill_mask = FILL_OUTLINE;
-    }
-  }
+  osd->fill_mask = osd->outline_offset ? FILL_OUTLINE :
+    osd->shadow_offset ? FILL_SHADOW : FILL_FACE;
 
   switch (osd->fill_mask) {
   case FILL_FACE:
@@ -320,10 +300,8 @@ expose_line(xosd * osd, int line)
     break;
   }
 
-  if (osd->fill_mask != FILL_NONE) {
-    XFillRectangle(osd->display, osd->line_bitmap, osd->gc,
-                   0, 0, osd->screen_width, osd->line_height);
-  }
+  XFillRectangle(osd->display, osd->line_bitmap, osd->gc,
+      0, 0, osd->screen_width, osd->line_height);
 
   /*
    * don't need to lock here because functions that call expose_line
